From 3635693d641739607ef7b0ba0abf3e172e339916 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 20 Jun 2013 20:32:50 +0530
Subject: [PORT FROM MAIN]TOMBSTONE at AudioFlinger::PlaybackThread::isOffloadTrack()

BZ: 61701

This patch will fix TOMBSTONE at AudioFlinger::PlaybackThread::
isOffloadTrack() during Google voice search.

Both PlaybackThread and RecorderThread uses EffectModule and hence
added a code to check the thread type before typecasting and calling
isOffloadTrack().

Change-Id: I9ea07a3026ca592340141220d99f3d03e7cd8eb8
Signed-off-by: samaga Prasanna Krishna <samaga.krishna@intel.com>
Orig-Change-Id: I27ada867230608a6b32364e4adca43d1fdfb4f57
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/Effects.cpp |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/services/audioflinger/Effects.cpp b/services/audioflinger/Effects.cpp
index e592410..959cb73 100644
--- a/services/audioflinger/Effects.cpp
+++ b/services/audioflinger/Effects.cpp
@@ -611,9 +611,10 @@ status_t AudioFlinger::EffectModule::setEnabled_l(bool enabled)
         if (thread == 0) {
             return NO_ERROR;
         }
-        PlaybackThread *p = (PlaybackThread *)thread.get();
 
-        if (enabled) {
+        if ((thread->type() == ThreadBase::DIRECT) && (enabled)) {
+            PlaybackThread *p = (PlaybackThread *)thread.get();
+            if (enabled) {
             if (p-> isOffloadTrack()) {
                 ALOGV("setEnabled: Offload, invalidate tracks");
                 DirectOutputThread *srcThread = (DirectOutputThread *)p;
-- 
1.7.4.1

